### ISSUE TITLE
Failed to convert property value of type 'org.springframework.security.web.access.intercept.FilterSecurityInterceptor' to required type 'javax.servlet.Filter'

### ISSUE THREAD

**Issue by guruscott-meyer** [grails-plugins/grails-spring-security-core#1013]
### Expected Behavior

The app should run and connect to Active Directory over LDAP

### Actual Behaviour

Error creating bean with name 'filterInvocationInterceptorDeregistrationBean': Initialization of bean failed; nested exception is org.springframework.beans.ConversionNotSupportedException: Failed to convert property value of type 'org.springframework.security.web.access.intercept.FilterSecurityInterceptor' to required type 'javax.servlet.Filter' for property 'filter'; nested exception is java.lang.IllegalStateException: Cannot convert value of type 'org.springframework.security.web.access.intercept.FilterSecurityInterceptor' to required type 'javax.servlet.Filter' for property 'filter': no matching editors or conversion strategy found

### Steps To Reproduce

Create a new grails project with embedded tomcat
Add grails spring security plugins to build.gradle
Declare authManager bean in resources.groovy:

@bean
AuthenticationManager authManager(BaseLdapPathContextSource contextSource) {
LdapBindAuthenticationManagerFactory factory = new LdapBindAuthenticationManagerFactory(contextSource);
factory.setUserDnPatterns("uid={0},DC=edapt,DC=local");
return factory.createAuthenticationManager();
}

Import spring-security-core, spring-security-web and spring-security-config in build.gradle
Attempt to run the app

### Environment Information

Windows 11, OpenJDK 17.0.11

### Example Application

(https://github.com/HarrisLOGIC/stella)

### Version

6.1.1

**Comment by puneetbehl** [comment]
It appears you might be using an incompatible version of the library. I think the class `org.springframework.security.web.access.intercept.FilterSecurityInterceptor` is not extending the class `javax.servlet.Filter`.  

NOTE: The link to example repository seems to be broken.

**Comment by guruscott-meyer** [comment]
Correct.  Not only has the org.springframework.security.web.access.intercept.FilterSecurityInterceptor class been deprecated for years, it was removed in the latest release.

**Comment by codeconsole** [comment]
@guruscott-meyer  That is not true. [FilterSecurityInterceptor](https://github.com/spring-projects/spring-security/blob/main/web/src/main/java/org/springframework/security/web/access/intercept/FilterSecurityInterceptor.java) was not removed.  It is only deprecated and should be around at least until Spring Security 7 is released which won't for another year from now. ~ November 2025.
